components/expression/expression-renderer.js

```diff
@@ -244,180 +413,341 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
+    .range()
+    .concat(d3.scale.category20b().range())
+    .concat(d3.scale.category20c().range());
+  var profileContent = this.svgProfile_.append('g')
+    .attr(
+      'width',
+      this.canvasWidth_ - this.PROFILE_MARGINS.LEFT - this.PROFILE_MARGINS.RIGHT
+    )
+    .attr(
+      'height',
+      this.profileHeight_ - this.PROFILE_MARGINS.TOP - this.PROFILE_MARGINS.BOTTOM
+    );
+
+  var line = d3.svg.line()
+    .x(function(data, i) {
+      return xScale(i);
+    })
+    .y(function(data) {
+      return yScale(data);
+    })
+    .interpolate('linear');
+  this.data.profiles.forEach(function(path, i) {
+    var geneIndex = this.data.profiles[i].row;
+    var pathColor = colors[genotet.utils.hashString(heatmapData.geneNames[geneIndex]) % 60];
+    this.data.profiles[i] = {
+      geneName: heatmapData.geneNames[geneIndex],
+      row: geneIndex,
+      color: pathColor
+    };
+    profileContent.append('path')
+      .attr('d', line(heatmapData.values[geneIndex]))
+      .attr('stroke', pathColor)
+      .attr('fill', 'none')
+      .on('mouseover', function(value) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathHover', this.data.profiles[i]);
+      }.bind(this))
+      .on('mouseout', function(value) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathUnhover', this.data.profiles[i]);
+      }.bind(this))
+      .on('click', function(value, i) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathClick', this.data.profiles[i]);
+      }.bind(this));
+  }, this);
 };
 
 /**
- * Computes the horizontal and vertical label sizes for the heatmap.
- * The results are stored into:
- *     this.geneLabelWidth,
- *     this.conditionLabelHeight
+ * Adds the expression profiles for the selected genes as line charts.
  * @private
  */
-genotet.ExpressionRenderer.prototype.getHeatmapLabelSizes_ = function() {
-  // TODO(liana): Computes the horizontal/vertical text margin allocated for
-  // the heatmap labels.
-  if (!this.data.options.showGeneLabels) {
-    this.geneLabelWidth_ = 0;
-  } else {
-    // To implement... replace dummy value.
-    this.geneLabelWidth_ = 30;
-  }
-  if (!this.data.options.showConditionLabels) {
-    this.conditionLabelHeight_ = 0;
-  } else {
-    // To implement... replace dummy value.
-    this.conditionLabelHeight_ = 50;
-  }
+genotet.ExpressionRenderer.prototype.addGeneProfile_ = function(geneIndex) {
+  var path = this.path_ = {
+    row: geneIndex
+  };
+  this.data.profiles.push(path);
+  this.drawGeneProfiles_();
 };
 
-/** @inheritDoc */
-genotet.ExpressionRenderer.prototype.resize = function() {
-  this.base.resize.call(this);
-  this.render();
+/**
+ * Adds the expression profiles for the selected genes as line charts.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.removeGeneProfile_ = function(geneIndex) {
+  var index = -1;
+  this.data.profiles.forEach(function(path, i) {
```

Comment:
Use a for loop and break when you find an element may be better in this case.

Or if you don't mind being a bit not efficient, but much cleaner, do
```js
var index = this.data.profiles.map(function(path) {
  return path.row
}).indexOf(geneIndex);
```